sdk/python/feast/pyspark/launchers/aws/emr_utils.py

```diff
@@ -165,15 +165,27 @@ def _sync_offline_to_online_step(
     }
 
 
+class EmrJobRef(NamedTuple):
+    """ EMR job reference. step_id can be None when using on-demand clusters, in that case each
+    cluster has only one step """
+
+    cluster_id: str
+    step_id: Optional[str]
+
+
+def _job_ref_to_str(job_ref: EmrJobRef) -> str:
+    return ":".join(["emr", job_ref.cluster_id, job_ref.step_id or ""])
+
+
 class JobInfo(NamedTuple):
+    job_ref: EmrJobRef
     job_type: str
-    cluster_id: str
-    step_id: str
-    table_name: str
     state: str
+    table_name: Optional[str]
+    output_file_uri: Optional[str]
```

Comment:
I know this is a little hacky to have retrieval-job-specific param in the generic `JobInfo` but i didn't want to build an entire class hierarchy because of this param (at least not yet)